core/auth/src/main/java/software/amazon/awssdk/auth/credentials/InstanceProfileCredentialsProvider.java

```diff
@@ -110,6 +112,7 @@ private InstanceProfileCredentialsProvider(BuilderImpl builder) {
                                      .profileName(profileName)
                                      .build();
         this.ec2MetadataDisableV1Resolver = Ec2MetadataDisableV1Resolver.create(profileFile, profileName);
+        this.ec2MetadataServiceTimeoutResolver = Ec2MetadataServiceTimeoutResolver.create(profileFile, profileName);
```

Comment:
Instead of introducing yet another resolver, should we make it more flexible to support new settings in the future? Something like, `Ec2MetadataResolver`? (I know this ship has sailed for `Ec2metadataDisableV1Resolver` :( ).

Reply:
We already have `Ec2MetadataConfigProvider` I have model all the config related to this class now